src/Network/OAuth/OAuth2/Compat.hs

```diff
@@ -44,6 +44,7 @@ data OAuth2 = OAuth2
     , oauth2AuthorizeEndpoint :: URIRef Absolute
     , oauth2TokenEndpoint :: URIRef Absolute
     , oauth2RedirectUri :: Maybe (URIRef Absolute)
+    , oauth2AppRoot :: Maybe Text
```

Comment:
I don't think we want to expand this type with fields not present in [`hoauth2`s type](https://hackage.haskell.org/package/hoauth2-2.8.1/docs/Network-OAuth-OAuth2.html#t:OAuth2). Some do we hope to drop compatibility with older versions and just use it directly. A change like this would block that.

Reply:
I am with you regarding this change, this was a quick hack to be able to use relative path as the app root.